packages/server/src/ClientConnection.ts

```diff
@@ -31,6 +31,8 @@ export class ClientConnection {
   // this map indicates whether a `Connection` instance has already taken over for incoming message for the key (i.e. documentName)
   private readonly documentConnections: Record<string, boolean> = {}
 
+  private readonly connections: Record<string, Connection> = {}
```

Comment:
I think we can use `documentConnections` for this functionality (and just change typing from `string, boolean` to `string, Connection`)